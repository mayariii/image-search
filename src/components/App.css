@import url('https://fonts.googleapis.com/css2?family=Work+Sans:wght@400;600;700&display=swap');
@import url('https://fonts.googleapis.com/css2?family=Merriweather:wght@400;700&display=swap');

html {
    overflow-x: hidden;
}

body {
    font-family: 'Work Sans',
    sans-serif;
}



.flex-container {
    display: flex;
    flex-direction: column;
}

.header-container {
    display: flex;
    justify-content: center;
    align-items: center;
}

.header-image {
    width: 300px;
}

h1 {
    font-family: 'Merriweather',
        serif;
    font-size: 3rem;
}

@media screen and (max-width: 768px) {
    .header-container {
        flex-direction: column;
        text-align: center;
    }
}
