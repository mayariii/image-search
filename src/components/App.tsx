import React from 'react';
import SearchForm from './SearchForm';


const App = () => {
    
    const onFormSubmit = () => {
        console.log("Get images from API")
}


    return (
     <div>
         <SearchForm onSubmit={onFormSubmit}/>
    </div>
    );
}


export default App;