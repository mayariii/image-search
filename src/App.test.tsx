import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect'
import App from './components/App';

test('app component renders correctly', () => {
    const AppComponent = renderer.create(<App />).toJSON();
    expect(AppComponent).toMatchSnapshot();
})


test('search input is blank', () => {
    render(<App />)
    const searchInput = screen.getByLabelText(/Search for images:/i)
    expect(searchInput).toHaveValue("")
})

