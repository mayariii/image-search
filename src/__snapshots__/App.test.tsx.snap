// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component renders correctly 1`] = `
<div
  className="flex-container"
>
  <header
    className="header-container"
  >
    <div>
      <h1>
        iMage the Photo Wizard
      </h1>
      <p>
        Enter a search term and I will conjure up some photos!
      </p>
    </div>
    <img
      alt="Wizard Animation"
      className="header-image"
      src="wizard.gif"
    />
  </header>
  <main>
    <div>
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-container"
        >
          <label
            className="form-item"
            htmlFor="searchInput"
          >
            Search for images:
          </label>
          <input
            className="form-item form-input-text"
            id="searchInput"
            onChange={[Function]}
            placeholder="Enter your search query"
            type="text"
            value=""
          />
          <button
            className="form-item form-btn-primary"
            type="submit"
          >
            Search
          </button>
        </div>
      </form>
    </div>
    <div
      className="image-list-container"
    />
  </main>
  <footer>
    <nav>
      <div
        className="numbers-container"
      />
      <p
        style={
          Object {
            "display": "none",
          }
        }
      >
        Current page: 
        1
      </p>
    </nav>
  </footer>
</div>
`;
